fuel-core/src/executor.rs

```diff
@@ -327,16 +346,90 @@ impl Executor {
                     let (mint, _): (Mint, _) = mint.into();
                     return Err(Error::NotSupportedTransaction(Box::new(mint.into())))
                 }
-            }
+            };
+        }
+
+        if let ExecutionKind::Production = execution_kind {
+            coinbase_tx.outputs_mut().push(Output::coin(
+                self.config.block_producer.coinbase_recipient,
+                execution_data.coinbase,
+                AssetId::BASE,
+            ));
+            // TODO: Use a linked list instead of a vector to do it faster.
+            block.transactions.insert(0, coinbase_tx.clone().into());
         }
 
-        // TODO: Compare calculated coinbase with existing
-        // TODO: Insert at the first slot in production mode `Mint` transaction with right values.
-        // TODO: Apply coinbase transaction
+        coinbase_tx = self.check_coinbase(
+            block_height as Word,
+            coinbase_tx,
+            Some(execution_data.coinbase),
+        )?;
+
+        let coinbase_id = coinbase_tx.id();
+        self.persist_outputs(
+            *block.header.height(),
+            &coinbase_id,
+            block_db_transaction,
+            &[],
+            coinbase_tx.outputs(),
+        )?;
+        execution_data.tx_status.insert(
+            0,
+            (
+                coinbase_id,
+                TransactionStatus::Success {
+                    block_id: Default::default(),
+                    time: *block.header.time(),
+                    result: None,
```

Comment:
The execution result is not available for `Mint` and `Create` transactions, so I make it `Option`.